src/components/MeetNotification/index.tsx

```diff
@@ -0,0 +1,187 @@
+    [startTime, formattedCurrentTime]
+  );
+
+  const diffTimeEnd = useMemo(
+    () =>
+      new Date(endTime).getTime() - new Date(formattedCurrentTime).getTime(),
+    [endTime, formattedCurrentTime]
+  );
+
+  const isTimerRun = useMemo(
+    () => diffTimeStart < fiveMinutes && diffTimeStart > sec,
+    [diffTimeStart]
+  );
+
+  const isMeetInProgress = useMemo(
+    () => diffTimeEnd > 0 && diffTimeEnd <= fiveMinutes,
+    [diffTimeEnd]
+  );
+
+  const isNotifOpen = useMemo(
+    () => isTimerRun || isMeetInProgress,
+    [isTimerRun, isMeetInProgress]
+  );
+
+  useEffect(() => {
+    const diffEnd = new Date(endTime).getTime() - new Date().getTime();
+
+    const initialTimer = isTimerRun
+      ? Math.floor(diffTimeStart / sec)
+      : Math.floor(diffEnd / sec);
+
+    const intervalId = setInterval(() => {
+      setTimer((prevTimer) => prevTimer - 1);
+    }, sec);
+
+    if (initialTimer > 0) {
+      setTimer(initialTimer);
+    }
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, []);
+
+  useEffect(() => {
+    const socket = io(
+      `${import.meta.env.VITE_REACT_APP_BASE_URL_SERVER}appointment`,
+      {
+        auth: {
+          token,
+        },
+        transports: ['websocket', 'polling'],
+      }
+    );
+
+    socket.on('appointment_update', handleAppointmentStarted);
+
+    return () => {
+      socket.disconnect();
+    };
+  }, []);
+
+  const handleAppointmentStarted = (data: Appointment) => {
+    if (data) {
+      dispatch(socketAppointmentActions.updateNextAppointment(data));
+    }
+  };
+
+  const getPatient = useMemo(() => {
+    const gender =
+      patient?.gender === female
+        ? `${t('Notification.Ms')}`
+        : `${t('Notification.Mr')}`;
+    return `${gender} ${patient?.lastName}`;
+  }, [patient]);
+
+  const getDoctor = useMemo(
+    () =>
+      `${t('Appointments.doctor')} ${
+        doctor.role === local ? remoteDoctor.lastName : localDoctor.lastName
+      }`,
+    [remoteDoctor, doctor, localDoctor]
+  );
+
+  const calculateTimer = useMemo(
+    () =>
+      (startTime: string): TimerResult => {
+        const deadline = new Date(startTime);
+        const diff = deadline.getTime() - new Date().getTime() || 0;
+        const timer = diff > 0 ? Math.floor(diff / MILLISEC_PER_SEC) : 0;
+        const minutes = Math.floor(timer / SEC_PER_MIN) % SEC_PER_MIN;
+        const seconds = Math.floor(timer % SEC_PER_MIN);
+
+        return { minutes, seconds };
+      },
+    [startTime]
+  );
+
+  const { minutes, seconds } = calculateTimer(startTime);
+
+  const getNotificationText = useMemo(() => {
+    return isTimerRun
+      ? `${t('Notification.youHaveVideoCallIn')}  ${`${minutes
+          .toString()
+          .padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`} minutes
+            with:`
+      : `${t('Notification.youCurrentlyHaveMeeting')}`;
+  }, [minutes, seconds]);
+
+  return (
```

Comment:
you have 162 lines of logic calculation in the component? don't you think its a good sign to create custom hook